clients/include/device_strided_batch_vector.hpp

```diff
@@ -0,0 +1,235 @@
+//
+// Copyright 2018-2019 Advanced Micro Devices, Inc.
+//
+#pragma once
+
+template <typename T>
+class host_strided_batch_vector;
+
+//!
+//! @brief Implementation of a strided batched vector on device.
+//!
+template <typename T, size_t PAD = 4096, typename U = T>
+class device_strided_batch_vector : public d_vector<T, PAD, U>
+{
+public:
+    //!
+    //! @brief The storage type to use.
+    //!
+    typedef enum class estorage
+    {
+        block,
+        interleave
+    } storage;
+
+    //!
+    //! @brief Disallow copying.
+    //!
+    device_strided_batch_vector(const device_strided_batch_vector&) = delete;
+
+    //!
+    //! @brief Disallow assigning.
+    //!
+    device_strided_batch_vector& operator=(const device_strided_batch_vector&) = delete;
+
+    //!
+    //! @brief Constructor.
+    //! @param n   The length of the vector.
+    //! @param inc The increment.
+    //! @param stride The stride.
+    //! @param batch_count The batch count.
+    //! @param stg The storage format to use.
+    //!
+    explicit device_strided_batch_vector(rocblas_int    n,
+                                         rocblas_int    inc,
+                                         rocblas_stride stride,
+                                         rocblas_int    batch_count,
+                                         storage        stg = storage::block)
+        : d_vector<T, PAD, U>(calculate_size(n, inc, stride, batch_count, stg))
+        , m_storage(stg)
+        , m_n(n)
+        , m_inc(inc)
+        , m_stride(stride)
+        , m_batch_count(batch_count)
+    {
+        bool valid_parameters = true;
+
+        switch(this->m_storage)
+        {
+        case storage::block:
+        {
+            if(this->m_stride < this->m_n * this->m_inc)
```

Comment:
so only positive stride for now. okay

Reply:
No, needs a correction. Good point.

Reply:
The raison d'être of this change, was to support negative strides during tests.

Reply:
absolute values are now used.